components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -253,6 +253,47 @@ fn sqrt(arg: &Option<Real>) -> Result<Option<Real>> {
     }))
 }
 
+#[inline]
+#[rpn_fn]
+pub fn asin(arg: &Option<Real>) -> Result<Option<Real>> {
+    Ok(arg.map_or(None, |arg| {
+        let arg = arg.into_inner();
+        if arg >= -1f64 && arg <= 1f64 {
```

Comment:
I guess we actually don't need to check the range, since "Return value is .... or NaN if the number is outside the range [-1, 1]." and NaN will be handled by `Real::new()` :)

Reply:
Yeah, you're right :)